lib/runner/extensions/event.command.js

```diff
@@ -28,6 +33,60 @@ var _ = require('lodash'),
             error.name = ASSERTION_FAILURE;
         }
         return error ? serialisedError(error, true) : undefined;
+    },
+
+    /**
+     * Removes files in Request body if any.
+     *
+     * @param {Request~definition} request Request JSON representation to be sanitized
+     * @param {Function} warn function to invoke when a file is found
+     *
+     * @returns {Request~definition}
+     */
+    sanitizeFiles = function (request, warn) {
+        var sanitizedRequest,
+            sanitizedFormdata;
+
+        // do nothing if request body is empty
+        if (!request.body) {
+            return request;
+        }
+
+        // handle request body in `file` mode
+        if (request.body.mode === REQUEST_BODY_MODE_FILE) {
+            // clone the request and remove file in body
+            sanitizedRequest = _.clone(request);
+            sanitizedRequest.body.file = null;
+
+            // send a warning that file was removed
+            warn();
+
+            return sanitizedRequest;
+        }
+
+        // handle request body in `formdata` mode
+        if (request.body.mode === REQUEST_BODY_MODE_FORMDATA) {
+            // extract any file type params
```

Comment:
I feel this segment could be made cleaner by:
1. Cloning the request.
2. Traversing the members of the cloned formdata request body.
3. Setting any encountered `src` values to null.

This ensures that at most one copy+traversal of the request/request-body is made. Currently, the request body has to be traversed once during `_.reject`, and then a second time via clone.